lib/ansible/modules/cloud/amazon/route53_info.py

```diff
@@ -252,7 +252,12 @@ def list_hosted_zones(client, module):
     if module.params.get('delegation_set_id'):
         params['DelegationSetId'] = module.params.get('delegation_set_id')
 
-    return client.list_hosted_zones(**params)
+    paginator = client.get_paginator('list_hosted_zones')
+    zones = [zone for page in paginator.paginate(**params) for zone in page['HostedZones']]
```

Comment:
You can avoid these list comprehensions by using the paginator's `.build_full_result()` method, which keeps things consistent with the other AWS modules and a little easier to read.
`zones = paginator.paginate().build_full_result()`

Reply:
ok, thanks! fixed!